syft/formats/spdx22json/to_format_model.go

```diff
@@ -53,32 +54,15 @@ func toPackages(catalog *pkg.Catalog, relationships []artifact.Relationship) []m
 	for _, p := range catalog.Sorted() {
 		license := spdxhelpers.License(p)
 		packageSpdxID := model.ElementID(p.ID()).String()
-		filesAnalyzed := false
-
-		// we generate digest for some Java packages
-		// see page 33 of the spdx specification for 2.2
-		// spdx.github.io/spdx-spec/package-information/#710-package-checksum-field
-		var checksums []model.Checksum
-		if p.MetadataType == pkg.JavaMetadataType {
-			javaMetadata := p.Metadata.(pkg.JavaMetadata)
-			if len(javaMetadata.ArchiveDigests) > 0 {
-				filesAnalyzed = true
-				for _, digest := range javaMetadata.ArchiveDigests {
-					checksums = append(checksums, model.Checksum{
-						Algorithm:     strings.ToUpper(digest.Algorithm),
-						ChecksumValue: digest.Value,
-					})
-				}
-			}
-		}
+
 		// note: the license concluded and declared should be the same since we are collecting license information
 		// from the project data itself (the installed package files).
 		packages = append(packages, model.Package{
-			Checksums:        checksums,
+			Checksums:        toPackageChecksums(p),
 			Description:      spdxhelpers.Description(p),
 			DownloadLocation: spdxhelpers.DownloadLocation(p),
 			ExternalRefs:     spdxhelpers.ExternalRefs(p),
-			FilesAnalyzed:    filesAnalyzed,
+			FilesAnalyzed:    false, // this directly relates to packageVerificationCode
```

Comment:
See the earlier comment. I believe this still holds that it can be true given that the file content of this package has been available for or subjected to analysis when creating the SPDX document.

Reply:
Thanks for the review @spiffcs -- the problem with `FilesAnalyzed`, which we were only setting `true` when providing Java checksums [here](https://github.com/anchore/syft/pull/1265/files#diff-e0a0d8eccb8d2ea7bdbff4619151963a07c5ed32de52a12407219def486c039dL65) is that according to the SPDX spec, [we now need to _also_ provide a _Package Verification Code_](https://spdx.github.io/spdx-spec/v2.3/package-information/#79-package-verification-code-field):
> Required | Yes
> 0..1 if FilesAnalyzed ([7.8](https://spdx.github.io/spdx-spec/v2.3/package-information/#7.8)) is true or omitted, 0..0 (must be omitted) if FilesAnalyzed is false.

I think I've misinterpreted this and there's a bug in the documentation that `PackageVerificationCode` is actually _not_ required.

But we should actually be returning `true` for everything -- is that your opinion?

Reply:
Offline conversation: FilesAnalyzed should actually be true in the one case it was, corrected!